scipy/stats/tests/test_survival.py

```diff
@@ -110,39 +153,110 @@ def test_nonunique(self):
-
-        sample = stats.CensoredData.right_censored(times, censored)
+        sample, times, censored = self.get_random_sample(rng, n_unique)
         res = stats.ecdf(sample)
         ref = _kaplan_meier_reference(times, censored)
         assert_allclose(res.x, ref[0])
-        assert_allclose(res.sf, ref[1])
+        assert_allclose(res.sf.points, ref[1])
 
         # If all observations are uncensored, the KM estimate should match
         # the usual estimate for uncensored data
         sample = stats.CensoredData(uncensored=times)
         res = _survival._ecdf_right_censored(sample)  # force Kaplan-Meier
         ref = stats.ecdf(times)
-        assert_equal(res.x, ref.x)
-        assert_allclose(res.cdf, ref.cdf, rtol=1e-14)
-        assert_allclose(res.sf, ref.sf, rtol=1e-14)
+        assert_equal(res[0], ref.x)
+        assert_allclose(res[1], ref.cdf.points, rtol=1e-14)
+        assert_allclose(res[2], ref.sf.points, rtol=1e-14)
+
+    def test_right_censored_ci(self):
+        # test "greenwood" confidence interval against example 5 (URL above).
+        times, died = self.t4, self.d4
+        sample = stats.CensoredData.right_censored(times, np.logical_not(died))
+        res = stats.ecdf(sample)
+        ref_allowance = [0.096, 0.096, 0.135, 0.162, 0.162, 0.162, 0.162,
+                         0.162, 0.162, 0.162, 0.214, 0.246, 0.246, 0.246,
+                         0.246, 0.341, 0.341]
+
+        sf_ci = res.sf.confidence_interval()
+        cdf_ci = res.cdf.confidence_interval()
+        allowance = res.sf.points - sf_ci.low
+
+        assert_allclose(allowance, ref_allowance, atol=1e-3)
+        assert_allclose(sf_ci.low, np.clip(res.sf.points - allowance, 0, 1))
+        assert_allclose(sf_ci.high, np.clip(res.sf.points + allowance, 0, 1))
+        assert_allclose(cdf_ci.low, np.clip(res.cdf.points - allowance, 0, 1))
+        assert_allclose(cdf_ci.high, np.clip(res.cdf.points + allowance, 0, 1))
+
+    def test_right_censored_exponential_greenwood_ci(self):
+        # test "exponential greenwood" confidence interval against example 5
+        times, died = self.t5, self.d5
+        sample = stats.CensoredData.right_censored(times, np.logical_not(died))
+        res = stats.ecdf(sample)
+        lower = np.array([0.66639, 0.624174, 0.456179, 0.287822, 0.287822,
+                          0.287822, 0.128489, 0.030957, 0.030957, 0.030957])
+        upper = np.array([0.991983, 0.970995, 0.87378, 0.739467, 0.739467,
+                          0.739467, 0.603133, 0.430365, 0.430365, 0.430365])
+
+        sf_ci = res.sf.confidence_interval(method='exponential greenwood')
+        cdf_ci = res.cdf.confidence_interval(method='exponential greenwood')
+
+        assert_allclose(sf_ci.low, lower, atol=1e-5)
+        assert_allclose(sf_ci.high, upper, atol=1e-5)
+        assert_allclose(cdf_ci.low, 1-upper, atol=1e-5)
+        assert_allclose(cdf_ci.high, 1-lower, atol=1e-5)
+
```

Comment:
Optionally, add test against Mathematica for example 4.

@tupui, Mathematica's default confidence interval is the exponential greenwood.

```Mathematica
e = {24, 3, 11, 19, 24, 13, 14, 2, 18, 17, 24, 21, 12, 1, 10, 23, 6, 5, 9, 17}
ci = {1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0}
R = EventData[e, ci]
S = SurvivalModelFit[R]
S["PointwiseIntervals", ConfidenceLevel->0.95, ConfidenceTransform->"LogLog"]
```

```{{0.694743,0.992802},{0.694743,0.992802},{0.647529,0.973299},{0.591142,0.947073},{0.591142,0.947073},{0.591142,0.947073},{0.591142,0.947073},{0.591142,0.947073},{0.591142,0.947073},{0.591142,0.947073},{0.464605,0.906422},{0.370359,0.856521},{0.370359,0.856521},{0.370359,0.856521},{0.370359,0.856521},{0.160489,0.776724},{0.160489,0.776724}} ```

Reply:
Done.